### PR TITLE
add linear shape functions for displacement in THM process

### PR DESCRIPTION
**Pull request by joergbuchwald**
analogous to pull request #2843 adding linear shape functions  for displacement to THM process.
https://github.com/ufz/ogs/pull/2843/

Comparison between quad and linear elements for this benchmark:
https://github.com/joergbuchwald/projects/blob/master/benchmarks/lin_vs_quad.ipynb

Fixes # 

1. [ ] Feature description was added to the [changelog](https://github.com/ufz/ogs/wiki/Release-notes-6.3.1)
2. [ ] Tests covering your feature were added?
3. [ ] Any new feature or behavior change was documented?

### CODE REVIEW

File: Tests/Data/ThermoHydroMechanics/Linear/Point_injection/square_1e2.prj [hunk 1]
```diff
@@ -293,6 +293,7 @@
                 <precon_type>ILUT</precon_type>
                 <max_iteration_step>10000</max_iteration_step>
                 <error_tolerance>1e-8</error_tolerance>
+                <scaling>1</scaling>
```
Comment (endJunction) [inline Tests/Data/ThermoHydroMechanics/Linear/Point_injection/square_1e2.prj]: Not used by BiCGSTAB. Only for SparseLU.

File: ProcessLib/ThermoHydroMechanics/LocalDataInitializer.h [hunk 2]
```diff
@@ -218,68 +301,86 @@ class LocalDataInitializer final
...
     // local assembler builder implementations.
 private:
-    /// Generates a function that creates a new LocalAssembler of type
-    /// LAData<ShapeFunctionDisplacement>. Only functions with shape function's
-    /// dimension less or equal to the global dimension are instantiated, e.g.
-    /// following combinations of shape functions and global dimensions: (Line2,
-    /// 1),
-    /// (Line2, 2), (Line2, 3), (Hex20, 3) but not (Hex20, 2) or (Hex20, 1).
-    template <typename ShapeFunctionDisplacement>
+        /// Generates a function that creates a new LocalAssembler of type
```
Comment (endJunction) [inline ProcessLib/ThermoHydroMechanics/LocalDataInitializer.h]: Formatting.
```sh
git clang-format your_ufz_remote/master
```

File: Tests/Data/ThermoHydroMechanics/Linear/Point_injection/square_1e2.prj (hunk 1 above)
Comment (joergbuchwald) [reply Tests/Data/ThermoHydroMechanics/Linear/Point_injection/square_1e2.prj]: This is also what I thought, but it makes a huge difference for convergence and run time:
with scaling:
info: === Time stepping at step #1 and time 5000 with step size 5000
info: Calculate non-equilibrium initial residuum.
info: [time] Assembly took 0.038145 s.
info: [time] Applying Dirichlet BCs took 0.00581064 s.
info: ------------------------------------------------------------------
info: *** Eigen solver computation
info: -> scale
info: -> solve with BiCGSTAB (precon ILUT)
info: 	 iteration: 1/10000
info: 	 residual: 6.772733e-09

info: ------------------------------------------------------------------
info: [time] Linear solver took 0.163631 s.
without: 
info: === Time stepping at step #1 and time 5000 with step size 5000
info: Calculate non-equilibrium initial residuum.
info: [time] Assembly took 0.037899 s.
info: [time] Applying Dirichlet BCs took 0.0050468 s.
info: ------------------------------------------------------------------
info: *** Eigen solver computation
info: -> solve with BiCGSTAB (precon ILUT)
info: 	 iteration: 369/10000
info: 	 residual: 6.246455e-09

info: ------------------------------------------------------------------
info: [time] Linear solver took 1.74722 s.

**Comment by endJunction** [comment]
sorry, I had to repush the master.

@joergbuchwald Please update the changelog.

**Pull request merged**